pkg/patterns/declarative/pkg/applier/type.go

```diff
@@ -19,6 +20,11 @@ type ApplierOptions struct {
 	Namespace  string
 	Validate   bool
 
+	CascadingStrategy metav1.DeletionPropagation
+
+	PruneWhitelist []string
```

Comment:
Nit: We probably should change this to something like `PruneResources` to avoid non-inclusive language (because this becomes part of our API I think, so renaming it later is harder)

Reply:
Totally agree! I have thought about this.
But, 
1. if we call it `PruneResources` we have a different type for the same field in `ApplierOptions` than in `ApplyOpts` which probably is confusing.
2. We could call it `PruneAllowList`, I have seen this somewhere. But then it is confusing as we go from the interface `PruneWhitelister` to `PruneAllowList` to `PruneResources`. 

This detail is so hard.
Maybe this is actually the time to change `PruneWhitelister` to `PruneAllowLister`? I believe it has never worked anyways up until now? Probably that's still not a cool thing to do though.

Reply:
Naming is always the hardest thing!  Let's do it as its own PR, so we can think about it in isolation (including the consequences in terms of breaking changes!)